pkg/cmd/roachtest/test_impl.go

```diff
@@ -499,14 +499,33 @@ func (t *testImpl) failureMsg() string {
 // target. If it does, `refError` is set to that target error value
 // and returns true. Otherwise, it returns false.
 func failuresMatchingError(failures []failure, refError any) bool {
+	// unwrap unwraps the error passed to find the deepest error in the
+	// chain that satisfies the `refError` provided.
+	unwrap := func(err error) bool {
+		var matched bool
+		for {
+			if isRef := errors.As(err, refError); !isRef {
+				break
+			}
+
+			matched = true
+			err = errors.Unwrap(err)
+			if err == nil {
+				break
```

Comment:
Or keep going? What if the next occurrence can be unwrapped?

Reply:
`errors.Unwrap(err)` returning `nil` means the `err` passed doesn't wrap any other error, so there's no "next occurrence". But maybe I misunderstand what you're trying to say.

Reply:
I was thinking of "multi-errors" (more in the comment for `UnwrapOnce`). Either way, `Unwrap` returns `nil`, in this case, so I suppose those errors aren't very likely.